### PR TITLE
Feat/cloudflare/dns_gateway

### PR DESCRIPTION
**Pull request by Adamowoc**
DNS logs part of the Cloudflare zerotrust parser. 

Related to : [38299](https://github.com/SekoiaLab/platform/issues/38299)

### CODE REVIEW

File: Cloudflare/zerotrust_dns/_meta/manifest.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+uuid: 7b1317ec-3f87-4b53-9b6d-3f79045f28fa
+name: ZeroTrust DNS
```
Comment (squioc) [inline Cloudflare/zerotrust_dns/_meta/manifest.yml:2]: Use the name of the dataset from Cloudflare
```suggestion
name: Cloudflare Gateway DNS
```

File: Cloudflare/zerotrust_dns/_meta/manifest.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+slug: zerotrust-dns
```
Comment (squioc) [inline Cloudflare/zerotrust_dns/_meta/manifest.yml:3]:
```suggestion
slug: cloudflare-gateway-dns
```

File: Cloudflare/zerotrust_dns/ingest/parser.yml [hunk 3]
```diff
@@ -0,0 +1,154 @@
+name: zerotrust-dns
```
Comment (squioc) [inline Cloudflare/zerotrust_dns/ingest/parser.yml:1]: To be coherent with other parsers for Cloudflare
```suggestion
name: Clouflare DNS Gateway
```

File: Cloudflare/zerotrust_dns/ingest/parser.yml [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+          256: "URI"
+          257: "CAA"
+          258: "AVC"
+          259: "DOA"
+          260: "AMTRELAY"
+          32768: "TA"
+          32769: "DLV"
+        mapping:
+          json_event.message.QueryType: dns.question.type
+        filter: '{{json_event.message.get("QueryTypeName") == None}}'
```
Comment (squioc) [inline Cloudflare/zerotrust_dns/ingest/parser.yml:123]: I think you want to test that `QueryTypeName`
```suggestion
        filter: '{{json_event.message.get("QueryTypeName") != None}}'
```

**Review by squioc — CHANGES_REQUESTED** [review]

File: Cloudflare/zerotrust_dns/ingest/parser.yml [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+          dns.question.type: '{{json_event.message.QueryTypeName}}'
+          event.category: ["network"]
+          event.kind: 'event'
+          event.type: ["info"]
+          user.email: '{{json_event.message.Email}}'
+          user.id: '{{json_event.message.UserID}}'
+          source.ip: '{{json_event.message.SrcIP}}'
+          source.port: '{{json_event.message.SrcPort}}'
+          network.protocol: 'dns'
+          network.transport: '{{json_event.message.Protocol}}'
```
Comment (squioc) [inline Cloudflare/zerotrust_dns/ingest/parser.yml:28]: Add some constant and easy fields:
```suggestion
          network.transport: '{{json_event.message.Protocol}}'
          event.dataset: "gateway_dns"
          observer.vendor: "Cloudflare"
          observer.type: "dns"
```

**Review by squioc — CHANGES_REQUESTED** [review]
I suggest changing the name of the format for 'gateway_dns'

File: Cloudflare/zerotrust_dns/ingest/parser.yml (hunk 4 above)
Comment (Adamowoc) [reply Cloudflare/zerotrust_dns/ingest/parser.yml:123]: I did meant to test that QueryTypeName did not exist. If it did exist, this step would be useless as dns.question.type would already be set (see line 19)

File: Cloudflare/gateway_dns/ingest/parser.yml [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+          21: "BADALG"
+          22: "BADTRUNC"
+          23: "BADCOOKIE"
+        mapping:
+          json_event.message.RCode: dns.response_code
+          
+  set_cloudflare_fields:
+    actions:
+      - set:
+          cloudflare: "{ {% for key, value in json_event.message.items() %}{% if value not in ['NA', ''] %}'{{key}}': r'{{value}}',{% endif %}{% endfor %} }"
```
Comment (Adamowoc) [inline Cloudflare/gateway_dns/ingest/parser.yml:157]: This line not working

**Review by squioc — DISMISSED** [review]

File: Cloudflare/gateway_dns/_meta/manifest.yml [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+name: Cloudflare Gateway DNS
+slug: cloudflare-gateway-dns
+
+description: >-
+  DNS queries inspected by Cloudflare Gateway
+
+data_sources:
+  Web logs: Log every event and request
+  DNS records: DNS Gateway logs
+  Authentication logs: Access requests are logged
```
Comment (squioc) [inline Cloudflare/gateway_dns/_meta/manifest.yml:11]: Only logs dns questions and answers. I suggest deleting others datasources.
```suggestion
  DNS records: DNS Gateway logs
```

**Review by squioc — APPROVED** [review]

**Review by squioc — APPROVED** [review]

**Pull request merged**